syntax = "proto3";

package virusscan;

service VirusScanService {
  rpc ScanFile (ScanFileRequest) returns (ScanFileResponse) {}
}

message ScanFileRequest {
  string file_name = 1;
  string file_SHA256 = 2;
  string file_SHA1 = 3;
  string file_MD5 = 4;
}

message ScanFileResponse {
  string file_name = 1;
  string file_SHA256 = 2;
  string file_SHA1 = 3;
  string file_MD5 = 4;
  bool is_infected = 5;
  string scan_result = 6;
  string scan_result_detail = 7;
}
